### ISSUE TITLE
Potential dependency conflicts between trains and urllib3

### ISSUE THREAD

**Issue by NeolithEra** [allegroai/trains#36]
Hi, _**trains**_ directly and transitively introduced multiple versions of urllib3.

As shown in the following full dependency graph of  **_trains_**,trains requires _**urllib3**_ (the latest version)， while the installed version of **_requests_**(2.22.0) requires _**urllib3>=1.21.1,<1.26**_.

According to Pip's “first found wins” installation strategy, _**urllib3 1.25.3**_ is the actually installed version.  

Although the first found package version _**urllib3 1.25.3**_ just satisfies the later dependency constraint (_**urllib3>=1.21.1,<1.26**_), it will lead to a build failure once developers release a newer version of urllib3.

### Dependency tree--------
```
trains(version range:)
| +-apache-libcloud(version range:>=2.2.1)
| +-attrs(version range:>=18.0)
| +-backports.functools-lru-cache(version range:>=1.0.2)
| +-boto3(version range:>=1.9)
| +-botocore(version range:>=1.12)
| +-enum34(version range:>=0.9)
| +-funcsigs(version range:>=1.0)
| +-furl(version range:>=2.0.0)
| | +-six(version range:>=1.8.0)
| | +-orderedmultidict(version range:>=1.0)
| +-future(version range:>=0.16.0)
| +-futures(version range:>=3.0.5)
| +-google-cloud-storage(version range:>=1.13.2)
| +-humanfriendly(version range:>=2.1)
| +-jsonmodels(version range:>=2.2)
| +-jsonschema(version range:>=2.6.0)
| +-numpy(version range:>=1.10)
| +-opencv-python(version range:>=3.2.0.8)
| +-pathlib2(version range:>=2.3.0)
| +-pigar(version range:>=0.9.2)
| | +-colorama(version range:>=0.3.9)
| | +-requests(version range:>=2.20.0)
| | | +-chardet(version range:>=3.0.2,<3.1.0)
| | | +-idna(version range:>=2.5,<2.9)
| | | +-urllib3(version range:>=1.21.1,<1.26)
| | | +-certifi(version range:>=2017.4.17)
| +-plotly(version range:>=3.9.0)
| +-psutil(version range:>=3.4.2)
| +-pyhocon(version range:>=0.3.38)
| | +-pyparsing(version range:>=2.0.3)
| +-pyjwt(version range:>=1.6.4)
| +-python-dateutil(version range:>=2.6.1)
| +-pyyaml(version range:>=3.12)
| +-requests(version range:>=2.20.0)
| | +-chardet(version range:>=3.0.2,<3.1.0)
| | +-idna(version range:>=2.5,<2.9)
| | +-urllib3(version range:>=1.21.1,<1.26)
| | +-certifi(version range:>=2017.4.17)
| +-requests-file(version range:>=1.4.2)
| +-six(version range:>=1.11.0)
| +-tqdm(version range:>=4.19.5)
| +-typing(version range:>=3.6.4)
| +-urllib3(version range:>=1.22)
```

Thanks for your attention.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution
1.	Fix your direct dependencies to be **_urllib3>=1.21.1,<1.26_** and **_requests==2.22.0_**, to remove this conflict. 
        I have checked this revision will not affect your downstream projects now.

2.	Remove your direct dependency urllib3, and use urllib3 transitively introduced by request.

@allegroai-git   Please let me know your choice. I can submit a PR to solve this issue.
Build a good dependency ecosystem for python projects is our common goal ^_^.

**Comment by bmartinn** [comment]
Hi @NeolithEra , Thank you for noticing the requirements conflict!

After discussing with @allegroai-git , the most straight forward solution is to lower the requirement of urllib3 to: urllib3 >= 1.21.1

My only comment on solution (1) is that we would not like to fix (or use upper limit) versions since we received numerous complains that it breaks users requirements installations.

From trains perspective, urllib3 1.21.1 works just fine (I guess the version limit was just inaccurate)

If you could submit a PR that would be greatly appreciated!  :)

Thanks!

**Comment by bmartinn** [comment]
Fixed in the latest release-candidate  0.10.5rc0
